### DISCUSSION TITLE
Capture jpg from mainstream

### DISCUSSION THREAD

**Discussion by ceewanna** [roflcoopter/viseron#682] [General]
All jpgs captured seem to have low resolution which I believe they must have come from the substream of cameras. Is there any setting to make viseron capture the motion jpg from the mainstream?

**Comment by roflcoopter** [comment]
You are correct, all the jpgs are from the substream. 
Viseron does not actually read frames from the mainstream when you have a substream configured, the mainstream is just saved to disk.

Just to clarify, are you talking about the thumbnails created when a recording starts or the MJPEG stream in the front end?

**Comment by ceewanna** [comment]
I am referring to the jpg motion snapshot with timestamp filename. I am wondering if it could be taken from the mainstream of which the resolution is far better.

**Reply by roflcoopter** [reply]
Then you need to remove the substream config, so Viseron runs detection on the main stream.
The problem is that there is no guarantee that the substream and main stream is in sync, so it is impossible to create accurate thumbnails from the mainstream if detection runs on the substream

**Reply by ceewanna** [reply]
Would running detection from the mainstream incur substantially higher cpu load?

**Reply by roflcoopter** [reply]
It will increase yes, but by how much is hard to say, easiest way is for you to just test and observe the processes
